base/intfuncs.jl

```diff
@@ -298,7 +298,11 @@ function invmod(n::T) where {T<:BitInteger}
 end
 
 # ^ for any x supporting *
-to_power_type(x) = convert(Base._return_type(*, Tuple{typeof(x), typeof(x)}), x)
+function to_power_type(x::Number)
+    T = promote_type(typeof(x), typeof(one(x)), typeof(x*x))
```

Comment:
```suggestion
    T = promote_type(typeof(x), typeof(x*x))
```

It just occured to me that a multiplicative identity doesn't necessarily exist:

https://en.wikipedia.org/wiki/Rng_(algebra)

Are such structures supported?

Reply:
I guess it'd be OK to require a type without a multiplicative identity to implement `^` themselves?

Reply:
This is breaking.

```
julia> using FixedPointNumbers

julia> (0.0Q0f15)^2
ERROR: ArgumentError: Q0f15 is a 16-bit type representing 65536 values from -1.0 to 0.99997; cannot represent 1
Stacktrace:
  [1] throw_converterror(::Type{Q0f15}, x::Int64)
    @ FixedPointNumbers ~/.julia/packages/FixedPointNumbers/Dn4hv/src/FixedPointNumbers.jl:326
  [2] rawone(::Type{Q0f15})
    @ FixedPointNumbers ~/.julia/packages/FixedPointNumbers/Dn4hv/src/fixed.jl:41
  [3] oneunit(::Type{Q0f15})
    @ FixedPointNumbers ~/.julia/packages/FixedPointNumbers/Dn4hv/src/FixedPointNumbers.jl:108
  [4] one(::Type{Q0f15})
    @ FixedPointNumbers ~/.julia/packages/FixedPointNumbers/Dn4hv/src/FixedPointNumbers.jl:109
  [5] one(x::Q0f15)
    @ FixedPointNumbers ~/.julia/packages/FixedPointNumbers/Dn4hv/src/FixedPointNumbers.jl:222
  [6] to_power_type(x::Q0f15)
    @ Base ./intfuncs.jl:336
```

Reply:
https://github.com/JuliaLang/julia/issues/57390